runtime/src/test_utils.rs

```diff
@@ -764,17 +764,26 @@ impl Runtime<MemoryBlockstore> for MockRuntime {
         self.balance.borrow().clone()
     }
 
-    fn resolve_address(&self, address: &Address) -> Option<Address> {
+    fn resolve_address(&self, address: &Address) -> Option<ActorID> {
         self.require_in_call();
-        if address.protocol() == Protocol::ID {
-            return Some(*address);
+        if let &Payload::ID(id) = address.payload() {
+            return Some(id);
+        }
+
+        match self.get_id_address(address) {
+            None => None,
+            Some(addr) => {
+                if let &Payload::ID(id) = addr.payload() {
```

Comment:
This can't fail, can it? Just unwrap()

Reply:
maybe for some tests `unable_to_resolve_client_address`, we should reserve this. How do you think?

Reply:
I agree we need the match block for `get_id_address`. If it fails to resolve, it should return none. But if it returns `Some` then the value _must_ be an ID address that we can unwrap.